vault/quotas/quotas_rate_limit_test.go

```diff
@@ -188,10 +188,16 @@ func TestRateLimitQuota_Allow_WithBlock(t *testing.T) {
 		require.NotZero(t, numFail)
 	}
 
-	// allow enough time for all blocked clients to be purged
-	time.Sleep(rlq.purgeInterval * 2)
-
-	for addr := range results {
-		require.False(t, rlq.hasBlockedClient(addr), "expected blocked client to be purged")
-	}
+	func() {
+		for {
+			select {
+			case <-time.After(rlq.purgeInterval):
```

Comment:
I don't really understand whether this works, sorry.

If we get a new `time.Tick(time.Second)` every time through the for loop, then don't we also get a new `time.After(rlq.purgeInterval)` too, pushing the deadline back every time?

I have not used `case <-time.Tick(time.Second):` as an idiom before.

Reply:
This will tick every second checking attempting to see if all clients have been garbage collected or it will timeout.

Reply:
Sure, but after you tick for one second, doesn't time.After(rlq.purgeInterval) also get re-run, extending the timeout?

Reply:
I've updated the function to create variables for the timeout and ticker. I've tested this by explicitly always returning non-zero in `numBlockedClients`. Feel free to unresolve this if you wish 👍